protocol/storage/deals_awaiting_seal.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/filecoin-project/go-filecoin/proofs/sectorbuilder"
+	"github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+func init() {
+	cbor.RegisterCborType(dealsAwaitingSeal{})
+}
+
+// sectorInfo combines sector Metadata from rust proofs with go-filecoin specific data
+type sectorInfo struct {
+	Metadata         *sectorbuilder.SealedSectorMetadata
+	CommitMessageCid *cid.Cid
```

Comment:
nit: does this need to be a pointer? We can open ourselves up to some [gotchas](https://github.com/filecoin-project/go-filecoin/blob/master/CONTRIBUTING.md#gotchas) with this. My preference would be to use a value and return `cid.Undef` inplace of `nil`.